oauth.go

```diff
@@ -111,6 +111,8 @@ type OAuthTokenParams struct {
 // OAuthToken is the value of the OAuthToken from OAuth flow.
 // https://stripe.com/docs/connect/oauth-reference#post-token
 type OAuthToken struct {
+	APIResource
```

Comment:
Is this an API resource? OAuth is a bit different and I'm worried the shape might cause problems (not sure if it's plausible)

Reply:
I think I see your meaning in that technically it's not an API resource, but it uses identical infrastructure to all the things that are for making the request, decoding, error handling, etc.:

``` go
func (c Client) New(params *stripe.OAuthTokenParams) (*stripe.OAuthToken, error) {
	// client_secret is sent in the post body for this endpoint.
	if stripe.StringValue(params.ClientSecret) == "" {
		params.ClientSecret = stripe.String(stripe.Key)
	}

	oauthToken := &stripe.OAuthToken{}
	err := c.B.Call(http.MethodPost, "/oauth/token", c.Key, params, oauthToken)

	return oauthToken, err
}
```

So the two alternatives would be:

1. Rename `APIResource` to something more generic so that it fits better with `OAuthToken`.
2. Have OAuth-related client calls use different request infrastructure.

I'm not too tied to `APIResource` — maybe a better name for now would be `ResourceWithLastRequest` or something, but couldn't think of a name that I liked any better. I also don't think it's too much big of a deal — if we add more API-specific stuff to `APIResource` in the future and it no longer fits with OAuth, we can always refactor OAuth at that time.

Reply:
I like APIResource personally because it represents a real API resource and it's a good way to find all "top-level classes" (without starting on a tangent I wish we had that at the API layer too in openapi)

I don't think we should change this I was flagging. cc @ob-stripe as we debated the same thing about OAuth in PHP last night

Reply:
Roger. Will leave as `APIResource` for now then.